query-service/schema/components/api-types.raml

```diff
@@ -0,0 +1,207 @@
+#%RAML 1.0 Library
+
+uses:
+  base: ./blast/blast-cli.raml
+
+types:
+  QueryJobRequest:
+    displayName: Query Job Request
+    type: object
+    additionalProperties: false
+    properties:
+      jobConfig:
+        displayName: Job Config
+        description: Job Configuration.
+        type: QueryJobConfig
+      blastConfig:
+        displayName: Blast Tool Config
+        description: |
+          Blast Tool Configuration.
+          
+          Implemented by one of:
+          
+          - [BlastNConfig](#type:lib.BlastNConfig)
+          - [BlastPConfig](#type:lib.BlastPConfig)
+          - [BlastXConfig](#type:lib.BlastXConfig)
+          - [DeltaBlastConfig](#type:lib.DeltaBlastConfig)
+          - [PSIBlastConfig](#type:lib.PSIBlastConfig)
+          - [RPSBlastConfig](#type:lib.RPSBlastConfig)
+          - [RPSTBlastNConfig](#type:lib.RPSTBlastNConfig)
+          - [TBlastNConfig](#type:lib.TBlastNConfig)
+          - [TBlastXConfig](#type:lib.TBlastXConfig)
+        type: base.BlastQueryConfig
+      meta:
+        displayName: User Metadata
+        type: QueryJobRequestMeta
+        required: false
+        default: null
+
+  QueryJobConfig:
+    type: object
+    additionalProperties: false
+    properties:
+      site:
+        displayName: Target Site
+        type: TargetSite
+      targets:
+        displayName: Query Targets
+        type: array
+        items:
+          type: QueryJobTarget
+        minItems: 1
+      query:
+        displayName: Blast Query
+        description: |
+          **For Requests**
+          
+          Inline query definition.
+          
+          This field is to be used for non-file-upload blast queries.
+          
+          If this field is not set, a query file must be uploaded for the
+          request to be valid.
+          
+          **For Responses**
+          
+          This field will be omitted.
+        type: string
+        required: false
+      addToCollection:
+        displayName: Add to Collection
+        description: |
+          Adds this job to the users collection of jobs.
+          
+          The user's collection of jobs is what is returned by the `/jobs`
+          endpoint.
+        type: boolean
+        required: false
+        default: false
+
+  QueryJobRequestMeta:
+    type: object
+    additionalProperties: false
+    properties:
+      summary:
+        displayName: Brief Summary
+        description: Brief, user defined summary or tagline for the job.
+        type: string
+        maxLength: 512
+        required: false
+      description:
+        displayName: Long Description
+        description: Longform, user provided description of the job.
+        type: string
+        required: false
+
+  QueryJobTarget:
+    type: object
+    additionalProperties: false
+    properties:
+      organism:
+        type: string
+      target:
```

Comment:
i am guessing this is `protein` | `NA` etc.  why not an enum?

Reply:
Discussed with Steve, will rename this field to `targetFile`

Reply:
Actually... maybe `organism` should be renamed `targetDisplayName`.  Organism is application specific, while multi-blast is more general. (EG, OrthoMCl does not have organisms). So we would have:
- `targetDisplayName`
- `targetFile`